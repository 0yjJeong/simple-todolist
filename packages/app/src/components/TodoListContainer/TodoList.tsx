import React from 'react';
import TodoList, { TodoListForm, useTodoList } from '@todolist/plugin-todo';
import ShadowBox from '../ShadowBox/ShadowBox';
import SlideToDeleteCard from '../../cards/SlideToDeleteCard';
import styled from 'styled-components';

const Wrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-left: 1rem;
  padding-right: 1rem;
  background-image: linear-gradient(to right, #ffecd2 0%, #fcb69f 100%);
`;

const TodoListContainer = () => {
  const { addTodo } = useTodoList();
  return (
    <Wrapper>
      <ShadowBox>
        <TodoListForm onAddTodo={addTodo} />
        <TodoList initialTodos={[]} CardComponent={SlideToDeleteCard} />
      </ShadowBox>
    </Wrapper>
  );
};

export default TodoListContainer;
