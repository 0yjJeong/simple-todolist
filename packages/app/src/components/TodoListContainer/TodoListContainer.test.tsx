import { render } from '@testing-library/react';
import 'jest-styled-components';
import TodoListContainer from './TodoList';

describe('<ShadowBox/>', () => {
  it('works', () => {
    const rendered = render(<TodoListContainer />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.getByPlaceholderText('할 일을 입력하세요')).toBeDefined();
  });
});
