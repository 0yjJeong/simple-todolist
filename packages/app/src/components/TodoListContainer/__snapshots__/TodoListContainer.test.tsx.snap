// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShadowBox/> works 1`] = `
.c1 {
  background: #ffffff;
  padding: 0.8rem;
  border-radius: 10px;
  height: calc(100% - 100px);
  box-shadow: 0 4px 9px 0 rgb(0 0 0 / 7%);
}

.c0 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 1rem;
  padding-right: 1rem;
  background-image: linear-gradient(to right,#ffecd2 0%,#fcb69f 100%);
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <form
      class="todolist__form"
    >
      <input
        placeholder="할 일을 입력하세요"
      />
      <button
        type="submit"
      >
        추가
      </button>
    </form>
    <ul
      class="todolist"
    />
  </div>
</div>
`;
